foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -210,8 +212,14 @@ def retry_lost_downloader_jobs() -> None:
                 # If this job really should be restarted we'll get it in the next loop.
                 if timezone.now() - job.created_at > MIN_LOOP_TIME:
                     lost_jobs.append(job)
+        except socket.timeout:
+            logger.info("Timeout connecting to Nomad - is Nomad down?", job_id=job.id)
+            lost_jobs.append(job)
+        except nomad.api.exceptions.BaseNomadException:
+            logger.info("Problem connecting to Nomad - is Nomad down?", job_id=job.id)
+            lost_jobs.append(job)
```

Comment:
I'm not sure that we should assume these jobs are lost for either of these except blocks. They could be happily chugging along while the only issue going on is that the Foreman is having network issues. I would think that better behavior would be to do nothing. If the jobs are actually in need of retrying, the Foreman will loop back over them in a couple minutes.